### PR TITLE
test: add coverage gate + service-handler tests

### PR DESCRIPTION
**Pull request by dewet22**
## Summary

First slice of the test-coverage improvements from a coverage analysis of the codebase. Tackles the two highest-leverage gaps: **no coverage measurement at all**, and the **untested service-handler layer** in `__init__.py`.

### Coverage gate
- Add `pytest-cov` to the dev group.
- Configure `[tool.coverage.*]` in `pyproject.toml`: branch coverage scoped to the integration package, `term-missing` report, and a conservative `fail_under` floor meant to be ratcheted upward as coverage improves.
- Run the `tests` CI job with `--cov --cov-report=term-missing` so regressions are caught and the real figure is visible in the log.

### Service-handler tests (`tests/test_init.py`)
Previously the service layer was the most under-tested surface relative to its risk (it sends real hardware commands). Added:
- `reboot_inverter` and `calibrate_battery_soc` — happy path (were entirely untested).
- Guard branches across the hardware-command services: unknown `device_id`, disconnected client (parametrized over reboot / calibrate / set_system_datetime).
- `set_system_datetime` — `serial`-target resolution and unknown-serial error path.
- `capture_frames` — unknown `device_id` and "no connected inverter" error paths.
- `expose_recommended_entities` — unknown `device_id` error path.

## Verification
- New tests pass locally (the sandbox needed a small `typing.ByteString` shim because only Python 3.14.0rc2 is installable there, not the project's required 3.14.2; CI runs on the real interpreter via the lockfile).
- Full suite reports **94%** coverage, comfortably above the floor.

## Follow-ups (remaining items from the analysis, not in this PR)
- Coordinator failure-recovery edge cases (mixed `ExceptionGroup`, `_reset_client` close-on-error, healed-callback exceptions).
- Config-flow `detect()` / `PlantTopologyMismatch` handling at setup time (likely a real fix, not just a test).
- `sensor.py` helper branches (`_battery_hex` exception/Enum paths) and the unasserted battery `status_*` / `warning_2` sensors.
- Migration-script I/O layer (`HAWebSocket`, `rebase_sum`, time helpers).
- Ratchet `fail_under` upward once CI reports the steady-state number.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Gp5kqvsPsWDKu2XSESMpA5

---
_Generated by [Claude Code](https://claude.ai/code/session_01Gp5kqvsPsWDKu2XSESMpA5)_

## Summary by CodeRabbit

* **Tests**
  * Implemented code coverage measurement with a 70% minimum threshold to ensure quality standards.
  * Added comprehensive integration tests for device commands, system configuration, and error handling scenarios.

### CODE REVIEW

File: tests/test_init.py [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 from unittest.mock import AsyncMock, MagicMock, patch
 
+import pytest
```
[the remark this answers is not in the document]
Comment (dewet22) [reply tests/test_init.py:5]: This is actually correct as-is and passes `ruff check` (including `--select I`). Ruff's isort, with the default `force-sort-within-sections = false`, separates plain `import` statements from `from` imports within each section and places the straight `import`s first — so `import pytest` belongs at the top of the third-party group, ahead of the `from ...` imports. Interleaving it alphabetically among the `from` imports (as suggested) is what would trigger the lint error. Verified locally: both `ruff check tests/test_init.py` and `ruff check --select I tests/test_init.py` report "All checks passed!".

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dewet22-codex** [comment]
I’ve started reviewing this PR.

File: pyproject.toml [hunk 2]
```diff
@@ -16,6 +16,7 @@ dependencies = [
 dev = [
     "pytest>=8.0",
     "pytest-asyncio>=0.23",
+    "pytest-cov>=5.0",
```
Comment (dewet22-codex) [inline pyproject.toml:19]: Adding `pytest-cov` here leaves the committed lock metadata stale because `uv.lock` is not part of this PR. I checked the PR head with `uv lock --check`, and it reports that the lockfile needs updating. `pytest-cov` already happens to be present transitively via the HA test plugin, so CI can still install and pass, but the root package dev-dependency metadata in `uv.lock` does not include this new direct dependency. I would include the `uv lock` update in this PR so local `uv lock --check` and future lock refreshes start from a consistent state.

Comment (dewet22) [reply pyproject.toml:19]: Good catch — fixed in 0b49433. `uv.lock` now records `pytest-cov` as a direct dev dependency in both `[package.dev-dependencies]` and `[package.metadata.requires-dev]` (specifier `>=5.0`). Resolution is unchanged since `pytest-cov` 7.1.0 and `coverage` were already locked transitively via `pytest-homeassistant-custom-component`, so this just adds the direct edge and brings `uv lock --check` back to a clean state.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dewet22-codex** [comment]
I’ve started re-reviewing this PR.

**Review by dewet22-codex — APPROVED** [review]
I’ve re-reviewed the latest head and have no concerns to add. The `uv.lock` metadata now includes `pytest-cov` as a direct dev dependency, and `uv lock --check` is clean on the PR head.
